src/app/study-filters/study-filters.component.ts

```diff
@@ -0,0 +1,101 @@
+import { Component, OnInit, Input, ViewChild, OnDestroy } from '@angular/core';
+import { Store } from '@ngxs/store';
+import { Dataset } from '../datasets/datasets';
+import { SetStudyFilters, StudyFiltersModel, StudyFiltersState } from './study-filters.state';
+import { StatefulComponent } from 'app/common/stateful-component';
+import { NgbNav } from '@ng-bootstrap/ng-bootstrap';
+import { StateReset } from 'ngxs-reset-plugin';
+import { SetNotEmpty } from 'app/utils/set.validators';
+import { Validate } from 'class-validator';
+import { DatasetNode } from 'app/dataset-node/dataset-node';
+import { DatasetsService } from 'app/datasets/datasets.service';
+import { findNodeById } from 'app/treelist-checkbox/treelist-checkbox.component';
+import { Subscription } from 'rxjs/internal/Subscription';
+
+@Component({
+  selector: 'gpf-study-filters-block',
+  templateUrl: './study-filters.component.html',
+  styleUrls: ['./study-filters.component.css']
+})
+export class StudyFiltersComponent extends StatefulComponent implements OnInit, OnDestroy {
+  @Input() public dataset: Dataset;
+
+  @Validate(SetNotEmpty, {message: 'Select at least one.'})
+  public selectedStudies: Set<string> = new Set<string>(['']); // Used by the child component(Treelist checkbox)
+  public allStudyNodes: DatasetNode[] = null;
+  public datasetSubscription: Subscription;
+
+  @ViewChild('nav') public ngbNav: NgbNav;
+
+  public constructor(protected store: Store, public datasets: DatasetsService) {
+    super(store, StudyFiltersState, 'studyFilters');
+  }
+
+  public showStudyFilters = false;
+
+  public ngOnInit(): void {
+    super.ngOnInit();
+    const datasets$ = this.datasets.getDatasetsObservable();
+    const datasetTrees = new Array<DatasetNode>();
+    this.datasetSubscription = datasets$.subscribe(datasets => {
+      datasets.filter(d => !d.parents.length).map(d => {
+        datasetTrees.push(new DatasetNode(d, datasets));
+      });
+      this.allStudyNodes = this.getAllStudyNodesById(datasetTrees, this.dataset.id);
+    });
+    this.store.selectOnce(state => state.studyFiltersState as StudyFiltersModel).subscribe(state => {
+      // restore state
+      if ((state?.studyFilters?.length ?? 0) !== 0) {
+        const timer = setTimeout(() => {
+          if (this.showStudyFilters === false) {
+            this.showStudyFilters = true;
+            this.ngbNav.select('studies');
+            this.updateSelectedStudies(new Set(state.studyFilters));
+            clearInterval(timer);
+          }
+        }, 200);
+      }
+    });
+  }
+
+  public ngOnDestroy(): void {
+    this.datasetSubscription.unsubscribe();
+  }
+
+  public onNavChange(): void {
+    if (this.ngbNav.activeId === 'studies') {
+      this.selectedStudies.add(''); // Otherwise the selected studies are going to be empty, preventing navigation
```

Comment:
Did you mean "preventing queries"? If yes, point out that it's because the validation for `selectedStudies` will be broken.